### PR TITLE
Ignore file ownership in git operations inside a container

### PR DESCRIPTION
**Pull request by jonatanschroeder**
Recent versions of git check the ownership of the repository for safety reasons. The issue is that, in local development using Docker containers, the owner of the course repos will typically be a user outside of the Docker container, which will trigger an error in any local git operation, since the owner of the directory doesn't match users in the container.

This PR disables this check by marking all directories as safe. This is done in the Dockerfile, so it only affects installations using Docker containers.

[As discussed on Slack](https://prairielearn.slack.com/archives/C0XBTPZMZ/p1694706440824859).

### CODE REVIEW

File: apps/prairielearn/src/lib/courseUtil.js [hunk 1]
```diff
@@ -14,7 +14,8 @@ function getCommitHash(coursePath, callback) {
     cwd: coursePath,
     env: process.env,
   };
-  exec('git rev-parse HEAD', execOptions, (err, stdout) => {
+  exec('git rev-parse HEAD', execOptions, (err, stdout, stderr) => {
+    console.log({ err, stdout, stderr });
```
Comment (nwalters512) [inline apps/prairielearn/src/lib/courseUtil.js:18]: Was this unintentionally committed?

Comment (jonatanschroeder) [reply apps/prairielearn/src/lib/courseUtil.js:18]: Sorry, this ended up being added in a rush. That said, adding the `stderr` to the error log may be useful to help debugging future issues of similar problems. Does it make sense to add the `stderr` content as the data for an `error.makeWithData` in the error handling?

Comment (nwalters512) [reply apps/prairielearn/src/lib/courseUtil.js:18]: Might be worth a shot! I'd recommend confirming that those properties are shown in the server job logs - I'm not sure off the top of my head if they will be or not.

Comment (jonatanschroeder) [reply apps/prairielearn/src/lib/courseUtil.js]: Done in e62c8bb40.

image

**Review by trombonekenny — APPROVED** [review]

**Review by nwalters512 — APPROVED** [review]

**Pull request merged**